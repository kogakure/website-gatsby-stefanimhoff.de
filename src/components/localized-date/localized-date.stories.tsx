import React from 'react';

import { LocalizedDate } from '.';

const date = '2020-01-17T18:00:00';

export default {
  component: LocalizedDate,
  title: 'Helpers/Localized Date',
};

export const Default = () => <LocalizedDate date={date} />;
