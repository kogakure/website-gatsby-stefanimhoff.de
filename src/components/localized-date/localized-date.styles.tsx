import styled from 'styled-components';

const LocalizedDateContainer = styled.time`
  cursor: pointer;
`;

const Kanji = styled.span`
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
`;

export const Styled = {
  LocalizedDateContainer,
  Kanji,
};
