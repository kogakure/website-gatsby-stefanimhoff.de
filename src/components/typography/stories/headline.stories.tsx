import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
