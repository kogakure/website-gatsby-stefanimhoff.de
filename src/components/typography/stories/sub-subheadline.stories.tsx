import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
