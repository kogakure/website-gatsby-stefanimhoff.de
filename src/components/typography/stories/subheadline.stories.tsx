import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
