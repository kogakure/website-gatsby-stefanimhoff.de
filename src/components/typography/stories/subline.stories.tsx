import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
