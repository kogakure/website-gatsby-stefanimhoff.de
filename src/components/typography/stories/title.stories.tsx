import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
