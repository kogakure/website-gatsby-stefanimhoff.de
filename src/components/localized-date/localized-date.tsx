import * as React from 'react';
import moment from 'moment';

import { LocalizedDateContext } from '../../contexts';

import { Styled } from './localized-date.styles';

export type LocalizedDateProps = {
  date: string;
};

export const LocalizedDate: React.FC<LocalizedDateProps> = ({
  date,
  ...props
}) => {
  const en = moment(date).format('MMMM Do, YYYY');
  const de = moment(date).format('DD.MM.YYYY');
  const [day, month, year] = de.split('.');
  const { localizedDate, setLocalizedDate } = React.useContext(
    LocalizedDateContext
  );
  const currentLanguage = localizedDate;
  let string;

  const toggleLanguage = () => {
    const possibleLanguages = ['en', 'de', 'ja'];
    const index = possibleLanguages.indexOf(currentLanguage);

    if (index >= 0 && index < possibleLanguages.length - 1) {
      const nextLanguage = possibleLanguages[index + 1];
      setLocalizedDate(nextLanguage);
    } else {
      setLocalizedDate('en');
    }
  };

  if (currentLanguage === 'de') {
    string = de;
  } else if (currentLanguage === 'ja') {
    string = (
      <>
        {year}
        <Styled.Kanji>年</Styled.Kanji>
        {month}
        <Styled.Kanji>月</Styled.Kanji>
        {day}
        <Styled.Kanji>日</Styled.Kanji>
      </>
    );
  } else {
    string = en;
  }

  return (
    <Styled.LocalizedDateContainer
      aria-label="Toggle date format"
      dateTime={date}
      onClick={toggleLanguage}
      onKeyUp={(e) => e.key === 'Enter' && toggleLanguage()}
      role="button"
      tabIndex={0}
      {...props}
    >
      {string}
    </Styled.LocalizedDateContainer>
  );
};
