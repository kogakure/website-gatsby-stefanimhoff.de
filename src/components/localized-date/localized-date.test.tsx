import React from 'react';
import { axe } from 'jest-axe';
import user from '@testing-library/user-event';

import { render } from '../../services/test-utils';

import { Default } from './localized-date.stories';

describe('LocalizedDate', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('changes the date text on click', async () => {
    const { container, findByText } = render(<Default />);
    user.click(await findByText(/January/i));
    expect(container.firstChild).toHaveTextContent('17.01.2020');
    expect(container.firstChild).toMatchSnapshot();

    user.click(await findByText(/17.01.2020/i));
    expect(container.firstChild).toHaveTextContent('2020年01月17日');
    expect(container.firstChild).toMatchSnapshot();

    user.click(await findByText(/年/i));
    expect(container.firstChild).toHaveTextContent('January 17th, 2020');
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
