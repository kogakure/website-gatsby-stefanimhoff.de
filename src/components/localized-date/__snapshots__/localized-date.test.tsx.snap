// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedDate changes the date text on Enter keyUp 1`] = `
.c0 {
  cursor: pointer;
}

<time
  aria-label="Toggle date format"
  class="c0"
  datetime="2020-01-17T18:00:00"
  role="button"
  tabindex="0"
>
  17.01.2020
</time>
`;

exports[`LocalizedDate changes the date text on Enter keyUp 2`] = `
<time
  aria-label="Toggle date format"
  class="sc-AxjAm jIaAfJ"
  datetime="2020-01-17T18:00:00"
  role="button"
  tabindex="0"
>
  2020
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    年
  </span>
  01
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    月
  </span>
  17
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    日
  </span>
</time>
`;

exports[`LocalizedDate changes the date text on click 1`] = `
.c0 {
  cursor: pointer;
}

<time
  aria-label="Toggle date format"
  class="c0"
  datetime="2020-01-17T18:00:00"
  role="button"
  tabindex="0"
>
  17.01.2020
</time>
`;

exports[`LocalizedDate changes the date text on click 2`] = `
<time
  aria-label="Toggle date format"
  class="sc-AxjAm jIaAfJ"
  datetime="2020-01-17T18:00:00"
  role="button"
  tabindex="0"
>
  2020
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    年
  </span>
  01
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    月
  </span>
  17
  .c0 {
  font-size: 0.7em;
  margin-left: 0.1em;
  margin-right: 0.1em;
}

<span
    class="c0"
  >
    日
  </span>
</time>
`;

exports[`LocalizedDate renders correctly 1`] = `
.c0 {
  cursor: pointer;
}

<time
  aria-label="Toggle date format"
  class="c0"
  datetime="2020-01-17T18:00:00"
  role="button"
  tabindex="0"
>
  January 17th, 2020
</time>
`;
